### PR TITLE
Created Prototype CLM Notebook, Also Created Some Additional Tutorials 

### PR DESCRIPTION
**Pull request by SeanSalvador2**
- Created Prototype Causal Language Modeling Notebook - trying different models for full finetuning. Worked successfully with small gpt2 model, working on LLama 2 (7b) and Mistral (7B) → ran into Memory issues. Still need to run with 2 GPUs and LoRA. *Note - Access to Llama 2 and Mistral Models must be requested on HuggingFace first. 
- Created a few more tutorial notebooks: CLM_Tutorial walks through HuggingFace’s Causal Language Modeling guide with additional notes. I added.  LLM_Model_Selection reviews the types of LLM Architecture and some of the most common LLM families and their uses. Loss_Tutorial is an incomplete notebook that explores appropriate loss functions for our task.

### CODE REVIEW

**Comment by SeanSalvador2** [comment]
Ready for your review - Cleaned up Training notebook a bit, got rid of excess code and successfully recreated the error "Expected all tensors to be on the same device, but found at least two devices, cuda:1 and cuda:0!"
